File: SpringBoot/Crud/src/main/java/com/crudDemo/service/CustomerService.java
package com.crudDemo.service;

import com.crudDemo.model.Customer;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface CustomerService {


    Customer save(Customer customer);

    List<Customer> findAll();

    Customer update(Customer customer);

    Boolean deleteById(String id);

    Customer findById(String id);

}

File: Node/HTTP/src/services/interfaces/CustomerService.ts
import { Customer } from "../../models/Customer";

export interface CustomerService {
  save(customer: Customer): Promise<Customer>;
  update(customer: Customer): Promise<Customer>;
  findById(customerId: string): Promise<Customer>;
  deleteById(customerId: string): Promise<void>;
  findAll(): Promise< Customer[]>;
}
